cluster-autoscaler/processors/provreq/injector.go

```diff
@@ -44,6 +46,68 @@ type ProvisioningRequestPodsInjector struct {
 	clock  clock.PassiveClock
 }
 
+// IsSupportedClass checks if the ProvisioningRequest is supported by the processor.
+func (p *ProvisioningRequestPodsInjector) IsSupportedClass(provisioningClassName string, supportedClasses []string) bool {
+	if len(supportedClasses) == 0 {
+		_, found := provisioningrequest.SupportedProvisioningClasses[provisioningClassName]
+		return found
+	}
+
+	for _, supportedClass := range supportedClasses {
+		if provisioningClassName == supportedClass {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsAvailableForProvisioning checks if the provisioning request is the correct state for processing and provisioning has not been attempted recently.
+func (p *ProvisioningRequestPodsInjector) IsAvailableForProvisioning(pr *provreqwrapper.ProvisioningRequest) bool {
+	conditions := pr.Status.Conditions
+	if apimeta.IsStatusConditionTrue(conditions, v1.Failed) || apimeta.IsStatusConditionTrue(conditions, v1.Provisioned) {
+		return false
+	}
+	provisioned := apimeta.FindStatusCondition(conditions, v1.Provisioned)
+	if provisioned != nil {
+		if provisioned.Status == metav1.ConditionFalse && provisioned.LastTransitionTime.Add(defaultRetryTime).Before(p.clock.Now()) {
+			return true
+		}
+		return false
+	}
+	return true
+}
+
+// UpdateProvisioningRequestCondition updates the condition of the ProvisioningRequest.
+func (p *ProvisioningRequestPodsInjector) UpdateProvisioningRequestCondition(pr *provreqwrapper.ProvisioningRequest, conditionType string, status metav1.ConditionStatus, failedMessage string) error {
+	var reason, message string
+	switch conditionType {
+	case v1.Failed:
+		klog.Errorf("Failed to get pods for ProvisioningRequest %v", pr.Name)
+		reason = provreqconditions.FailedToCreatePodsReason
+		message = failedMessage
+	case v1.Accepted:
+		reason = provreqconditions.AcceptedReason
+		message = provreqconditions.AcceptedMsg
+	default:
+		klog.Errorf("Condition %s is not supported", conditionType)
+		return errors.New("unsupported condition")
+	}
+
+	provreqconditions.AddOrUpdateCondition(pr, conditionType, status, reason, message, metav1.NewTime(p.clock.Now()))
+	if _, err := p.client.UpdateProvisioningRequest(pr.ProvisioningRequest); err != nil {
+		klog.Errorf("Failed to add condition %s to ProvReq %s/%s, err: %v", conditionType, pr.Namespace, pr.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+// ListPodsForProvisioningRequest returns the list of pods for the ProvisioningRequest.
+func (p *ProvisioningRequestPodsInjector) ListPodsForProvisioningRequest(pr *provreqwrapper.ProvisioningRequest) ([]*apiv1.Pod, error) {
```

Comment:
What is the purpose of this method?

Reply:
The purpose was to separate the logic for listing Pods and marking PR conditions.